src/response_decoder.ts

```diff
@@ -288,13 +295,19 @@ export class ResponseDecoder {
       } else if (isDeliverResponse(response)) {
         this.emitter.emit("deliver", new DeliverResponse(response))
         this.logger.debug(`deliver received from the server: ${inspect(response)}`)
+      } else if (isCreditResponse(response)) {
+        this.emitter.emit("credit_response", new CreditResponse(response))
+        this.logger.debug(`credit received from the server: ${inspect(response)}`)
```

Comment:
I suggest making the log and the raise the event, this event is synchronous so in the log, if you switch it it's more readeble